airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/unstructured_parser.py

```diff
@@ -83,11 +90,25 @@ def parse_records(
     ) -> Iterable[Dict[str, Any]]:
         format = _extract_format(config)
         with stream_reader.open_file(file, self.file_read_mode, None, logger) as file_handle:
-            markdown = self._read_file(file_handle, file, format, logger)
-            if markdown is not None:
+            try:
+                markdown = self._read_file(file_handle, file, format, logger)
+                if markdown is not None:
+                    yield {
+                        "content": markdown,
+                        "document_key": file.uri,
+                        "error": None,
+                    }
+            except RecordParseError as e:
+                # pass explicitly raised RecordParseError through to be handled by the caller
+                raise e
+            except Exception as e:
+                exception_str = str(e)
+                logger.warn(f"File {file.uri} caused an error during parsing: {exception_str}.")
```

Comment:
Which exception types are raised and which are ignored seems reversed to me, but I may be missing context. The way I interpret our desired 'skip' behavior would when we run into a specific file which cannot be parsed, and for that I'd assume the `_read_file()` method would send a `RecordParseError`. 

The way we have it now, it seems an error like a network failure could caught in the second bucket and skipped (en masse) if there were some kind of intermittent network issue during the sync.

Reply:
RecordParseError will be raised if a file isn't parseable because it has the wrong file type and the "skip unprocessable file types" setting is not set.

All other errors will be treated as you say - they make it to the destination without content with the "error" column set. This could also happen on an intermittent network issue.

What do you think about downloading the file outside of this try/except block and only catch all the exceptions thrown during partitioning via unstructured?

Still leaves the question from your other comment, but these seem kind of separate to me